src/System.Management.Automation/engine/parser/PSType.cs

```diff
@@ -629,9 +630,19 @@ private PropertyBuilder EmitPropertyIl(PropertyMemberAst propertyMemberAst, Type
                 // The property set and property get methods require a special set of attributes.
                 var getSetAttributes = Reflection.MethodAttributes.SpecialName | Reflection.MethodAttributes.HideBySig;
                 getSetAttributes |= propertyMemberAst.IsPublic ? Reflection.MethodAttributes.Public : Reflection.MethodAttributes.Private;
-                if (ShouldImplementProperty(propertyMemberAst.Name, type))
+                MethodInfo implementingGetter = null;
+                MethodInfo implementingSetter = null;
+                if (ShouldImplementProperty(propertyMemberAst.Name, type, out PropertyInfo interfaceProperty))
                 {
-                    getSetAttributes |= Reflection.MethodAttributes.Virtual;
+                    if (propertyMemberAst.IsStatic)
+                    {
+                        implementingGetter = interfaceProperty.GetGetMethod();
+                        implementingSetter = interfaceProperty.GetSetMethod();
```

Comment:
Can these be protected? If so we may need `GetSetMethod(true)` and then check `IsFamily | IsFamilyOrAssembly | IsPublic`

Reply:
Technically yes they can be but the [ShouldImplementProperty](https://github.com/PowerShell/PowerShell/blob/9b93241d0ddea729d2d4ef331322ebfa054dd2ae/src/System.Management.Automation/engine/parser/PSType.cs#L635) method currently does not detect properties that are not public. I believe we should hold off on supporting that in this PR because there are open questions around when a property has mixed accessor for get/set.

At least for now this code is only called when the static interface property is public and won't be when protected.

Reply:
Yeah makes sense, we'll hold off then